src/main/kotlin/com/viartemev/thewhiterabbit/channel/Channels.kt

```diff
@@ -4,44 +4,57 @@ import com.rabbitmq.client.Channel
 import com.rabbitmq.client.Connection
 import com.viartemev.thewhiterabbit.channel.Channels.localChannels
 import com.viartemev.thewhiterabbit.channel.Channels.localConfirmChannels
+import com.viartemev.thewhiterabbit.channel.Channels.localTxChannels
 import com.viartemev.thewhiterabbit.consumer.ConfirmConsumer
 import com.viartemev.thewhiterabbit.publisher.ConfirmPublisher
+import com.viartemev.thewhiterabbit.publisher.TxPublisher
 import java.util.concurrent.ConcurrentHashMap
 import kotlin.concurrent.thread
 
 object Channels {
-    internal val localConfirmChannels = ConcurrentHashMap<Thread, UncloseableConfirmChannel>()
     internal val localChannels = ConcurrentHashMap<Thread, UncloseableChannel>()
+    internal val localConfirmChannels = ConcurrentHashMap<Thread, UncloseableConfirmChannel>()
+    internal val localTxChannels = ConcurrentHashMap<Thread, UnclosableTxChannel>()
 
     init {
         Runtime.getRuntime().addShutdownHook(thread(start = false) {
-            sequenceOf(localChannels.values.asSequence(), localConfirmChannels.values.asSequence())
+            sequenceOf(
+                localChannels.values.asSequence(),
+                localConfirmChannels.values.asSequence(),
+                localTxChannels.values.asSequence()
+            )
                 .flatten()
                 .forEach { it.close0() }
             localChannels.clear()
             localConfirmChannels.clear()
+            localTxChannels.clear()
         })
     }
 
-
     private interface IAmUncloseableChannel : Channel {
         fun close0()
     }
 
     internal class UncloseableChannel(private val channel: Channel) : IAmUncloseableChannel, Channel by channel {
         override fun close() {}
-
         override fun close(closeCode: Int, closeMessage: String?) {}
 
         override fun close0() {
             if (channel.isOpen) channel.close()
         }
     }
 
-    internal class UncloseableConfirmChannel(private val channel: ConfirmChannel) : ConfirmChannel(channel),
-        IAmUncloseableChannel {
+    internal class UncloseableConfirmChannel(private val channel: ConfirmChannel) : ConfirmChannel(channel), IAmUncloseableChannel {
         override fun close() {}
+        override fun close(closeCode: Int, closeMessage: String?) {}
```

Comment:
Why did it appear here? I thought rabbitmq will use this method and will really close our channel which i needed on application exit. Am iI wrong?

Reply:
Actually it had been there, I just added `UnclosableTxChannel` class. All those `Unclosable*` are identical, so we'd better merge them. BTW what's the reason to have that channel's wrappers?

Reply:
Easy: for user to not be able to close channel while other fibers use it :)